### ISSUE TITLE
String comparison doesn't work in both SQLCE and SQL Server databases

### ISSUE THREAD

**Issue by MarcoViglione-Laser** [OrchardCMS/Orchard#5812]
I created a record repository to directly access the records of my table.

    private readonly IRepository<MyRecord> _myRecordRepository;

Then I'm trying to execute a LINQ query on the content of my table.

    var myQuery = _myRecordRepository.Table.Where(t.Message == myMessage);

The field called Message is defined in the migration as `.Column<string>("Message", col => col.Unlimited())` because I need to save messages longer than 255 characters. If I use SQLCE the migration generates this field in the database as an ntext, but my LINQ query gives me an error if I try to execute it because it cannot compare an ntext and a string. To have it work correctly I need to modify my query adding a ToString() like this:

    var myQuery = _myRecordRepository.Table.Where(t.Message.ToString() == myMessage);

And with SQLCE everything now works. Then I tried to enable my module on a tenant that instead uses a SQL Server database. In this tenant the migration generates the Message column as an nvarchar(MAX). I tried to execute the LINQ query but the output was not what I expected. By logging the queries executed by Orchard I found that the problem is caused by t.Message.ToString(), that when automatically translated by NHibernate becomes a `CAST(Message AS NVARCHAR(50))`. If instead I only use t.Message this cast is not generated.

So, long story short, if I create a column with unlimited size the same code cannot work under both SQLCE and SQL Server, because the former requires a ToString() instruction while the latter truncates the text if I use ToString().

**Comment by sebastienros** [comment]
This is a SQL limitation. you should use `Contains` for unlimited strings.
